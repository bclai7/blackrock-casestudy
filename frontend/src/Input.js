import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Form from 'react-bootstrap/Form';

function Input({handleChange, formData}) {
  return (
    <div className="Input" data-testid="userInput">
      <div className='inputHeader sectionHeader'>
        <Row>
          <Col>
            Inputs
          </Col>
        </Row>
      </div>
      
      <div className='inputFields'>
        <Row>
          <Col xs={{span: 1}}>
            <Form.Group className="mb-3" controlId="inputFormX" data-testid="x">
                <Form.Label>x0</Form.Label>
                <Form.Control type="number" name="x" placeholder="Value" value={formData.x} onChange={(e) => handleChange(e)} required/>
              </Form.Group>
          </Col>
          <Col xs={{span: 1}}>
            <Form.Group className="mb-3" controlId="inputFormY" data-testid="y">
                <Form.Label>y0</Form.Label>
                <Form.Control type="number" name="y" placeholder="Value" value={formData.y} onChange={(e) => handleChange(e)} required/>
              </Form.Group>
          </Col>
          <Col xs={{span: 1}}>
            <Form.Group className="mb-3" controlId="inputFormZ" data-testid="z">
                <Form.Label>z0</Form.Label>
                <Form.Control type="number" name="z" placeholder="Value" value={formData.z} onChange={(e) => handleChange(e)} required/>
              </Form.Group>
          </Col>
          <Col xs={{span: 1}}>
            <Form.Group className="mb-3" controlId="inputFormSigma" data-testid="sigma">
                <Form.Label>Sigma</Form.Label>
                <Form.Control type="number" name="sigma" placeholder="Value" value={formData.sigma} onChange={(e) => handleChange(e)} required/>
              </Form.Group>
          </Col>
          <Col xs={{span: 1}}>
            <Form.Group className="mb-3" controlId="inputFormRho" data-testid="rho">
                <Form.Label>Rho</Form.Label>
                <Form.Control type="number" name="rho" placeholder="Value" value={formData.rho} onChange={(e) => handleChange(e)} required/>
              </Form.Group>
          </Col>
          <Col xs={{span: 1}}>
            <Form.Group className="mb-3" controlId="inputFormBeta" data-testid="beta">
                <Form.Label>Beta</Form.Label>
                <Form.Control type="number" name="beta" placeholder="Value" value={formData.beta} onChange={(e) => handleChange(e)} required/>
              </Form.Group>
          </Col>
          <Col xs={{span: 1}}>
            <Form.Group className="mb-3" controlId="inputFormDelta" data-testid="delta">
                <Form.Label>Delta t</Form.Label>
                <Form.Control type="number" name="delta" placeholder="Value" value={formData.delta} onChange={(e) => handleChange(e)} required/>
              </Form.Group>
          </Col>
        </Row>
      </div>
    </div>
  );
}

export default Input;
