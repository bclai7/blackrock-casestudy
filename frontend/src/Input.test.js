import { render, screen, cleanup } from '@testing-library/react';
import Input from './Input';
import renderer from 'react-test-renderer';

test('renders Input component', () => {
  const formData = {
    x: '',
    y: '',
    z: '',
    sigma: '',
    rho: '',
    beta: '',
    delta: '',
  };
  render(<Input formData={formData}/>);
  const inputElement = screen.getByTestId('userInput');
  const xElement = screen.getByTestId('x');
  const yElement = screen.getByTestId('y');
  const zElement = screen.getByTestId('z');
  const sigmaElement = screen.getByTestId('sigma');
  const rhoElement = screen.getByTestId('rho');
  const betaElement = screen.getByTestId('beta');
  const deltaElement = screen.getByTestId('delta');
  expect(inputElement).toBeInTheDocument();
  expect(xElement).toBeInTheDocument();
  expect(yElement).toBeInTheDocument();
  expect(zElement).toBeInTheDocument();
  expect(sigmaElement).toBeInTheDocument();
  expect(rhoElement).toBeInTheDocument();
  expect(betaElement).toBeInTheDocument();
  expect(deltaElement).toBeInTheDocument();
});

test('input snapshot', () => {
    const formData = {
        x: '',
        y: '',
        z: '',
        sigma: '',
        rho: '',
        beta: '',
        delta: '',
      };
    const snap = renderer.create(<Input formData={formData}/>).toJSON();
    expect(snap).toMatchSnapshot();
});
