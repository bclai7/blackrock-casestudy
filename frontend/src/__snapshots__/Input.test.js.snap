// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input snapshot 1`] = `
<div
  className="Input"
  data-testid="userInput"
>
  <div
    className="inputHeader sectionHeader"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        Inputs
      </div>
    </div>
  </div>
  <div
    className="inputFields"
  >
    <div
      className="row"
    >
      <div
        className="col-1"
      >
        <div
          className="mb-3"
          data-testid="x"
        >
          <label
            className="form-label"
            htmlFor="inputFormX"
          >
            x0
          </label>
          <input
            className="form-control"
            id="inputFormX"
            name="x"
            onChange={[Function]}
            placeholder="Value"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        className="col-1"
      >
        <div
          className="mb-3"
          data-testid="y"
        >
          <label
            className="form-label"
            htmlFor="inputFormY"
          >
            y0
          </label>
          <input
            className="form-control"
            id="inputFormY"
            name="y"
            onChange={[Function]}
            placeholder="Value"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        className="col-1"
      >
        <div
          className="mb-3"
          data-testid="z"
        >
          <label
            className="form-label"
            htmlFor="inputFormZ"
          >
            z0
          </label>
          <input
            className="form-control"
            id="inputFormZ"
            name="z"
            onChange={[Function]}
            placeholder="Value"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        className="col-1"
      >
        <div
          className="mb-3"
          data-testid="sigma"
        >
          <label
            className="form-label"
            htmlFor="inputFormSigma"
          >
            Sigma
          </label>
          <input
            className="form-control"
            id="inputFormSigma"
            name="sigma"
            onChange={[Function]}
            placeholder="Value"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        className="col-1"
      >
        <div
          className="mb-3"
          data-testid="rho"
        >
          <label
            className="form-label"
            htmlFor="inputFormRho"
          >
            Rho
          </label>
          <input
            className="form-control"
            id="inputFormRho"
            name="rho"
            onChange={[Function]}
            placeholder="Value"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        className="col-1"
      >
        <div
          className="mb-3"
          data-testid="beta"
        >
          <label
            className="form-label"
            htmlFor="inputFormBeta"
          >
            Beta
          </label>
          <input
            className="form-control"
            id="inputFormBeta"
            name="beta"
            onChange={[Function]}
            placeholder="Value"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        className="col-1"
      >
        <div
          className="mb-3"
          data-testid="delta"
        >
          <label
            className="form-label"
            htmlFor="inputFormDelta"
          >
            Delta t
          </label>
          <input
            className="form-control"
            id="inputFormDelta"
            name="delta"
            onChange={[Function]}
            placeholder="Value"
            required={true}
            type="number"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
